common/src/main/java/com/wynntils/models/players/PlayerModel.java

```diff
@@ -192,8 +194,13 @@ private void clearUserCache() {
     }
 
     private void clearTextureCache() {
-        cosmeticTextures.forEach((uuid, array) -> Arrays.stream(array)
-                .forEach(l -> McUtils.mc().getTextureManager().release(l)));
+        for (ResourceLocation[] locations : cosmeticTextures.values()) {
+            // Make sure texture unloading is done on client thread
+            Arrays.stream(locations).forEach(l -> {
+                Managers.TickScheduler.scheduleNextTick(
```

Comment:
Do you need to schedule one call per location, or could you move the forEach into the scheduled lambda?

Reply:
whoops, good point